packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -283,6 +300,10 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
     const action = permission.action ?? 'lambda:InvokeFunction';
     const scope = permission.scope ?? this;
 
+    if (['lambda:InvokeFunction', 'lambda:*'].includes(action) && !qualified) {
```

Comment:
How do we feel about this? It's not the most elegant but I can't think of any other valid and sensible configuration that also catches `lambda:InvokeFunction`. I guess also `*`?

Reply:
Can you do the change for Resource based policies in a separate PR? It will be easier to evaluate separately.